tests/infra/remote.py

```diff
@@ -278,12 +278,23 @@ def wait_for_stdout_line(self, line, timeout):
                 if stdout.channel.recv_exit_status() == 0:
                     return
                 time.sleep(1)
-            raise ValueError(
-                "{} not found in stdout after {} seconds".format(line, timeout)
-            )
+            raise ValueError(f"{line} not found in stdout after {timeout} seconds")
         finally:
             client.close()
 
+    def check_for_stdout_line(self, line, timeout):
+        try:
+            client = self._connect_new()
+            for _ in range(timeout):
+                _, stdout, _ = client.exec_command(f"grep -F '{line}' {self.root}/out")
```

Comment:
We should use `self.out`, rather than reconstructing it here.

Reply:
Minor: I don't think these functions/variables should be called `_line`, since we're just grepping for a string rather than an entire line?

Reply:
line suggests a single line, but then -F suggests a string with line returns, am I missing something?

Reply:
I guess they could be named something like `check/wait_stdout_expression` even though I think that `_line` does the job with no ambiguity as to what the function is doing.

Reply:
> line suggests a single line, but then -F suggests a string with line returns, am I missing something?

`-F` just tells grep to treat the pattern as an exact string (ie - not a regex pattern), in neither case are we matching against entire lines.

Reply:
In both cases we are trying to find a string in one of the lines in stdout, hence (I am guessing) the original naming, as in wait for this string to appear in a line in stdout. If there are no good naming alternatives I will leave it as is for now.